OpenRA.Mods.Common/SpriteLoaders/GenieSlpLoader.cs

```diff
@@ -33,20 +49,30 @@ public GenieSlpFrame(Stream stream, GenieSlpFrameHeader header)
 				Size = FrameSize = new Size(header.Width, header.Height);
 				Offset = float2.Zero;
 
-				stream.Position = header.CommandTableOffset;
+				var skipLeft = new ushort[header.Height];
+				var skipRight = new ushort[header.Height];
+				for (var i = 0; i < header.Height; i++)
+				{
+					skipLeft[i] = stream.ReadUInt16();
+					skipRight[i] = stream.ReadUInt16();
+				}
 
 				var frameData = new List<byte[]>();
 
 				for (var y = 0; y < header.Height; y++)
-					frameData.Add(ReadRowCommands(stream, header));
+				{
+					stream.Position = header.RowCommandOffsets[y];
+					frameData.Add(ReadRowCommands(stream, header, skipLeft[y]));
+				}
 
+				Data = frameData.SelectMany(x => x).ToArray();
 			}
 
-			static byte[] ReadRowCommands(Stream stream, GenieSlpFrameHeader header)
+			static byte[] ReadRowCommands(Stream stream, GenieSlpFrameHeader header, ushort skipLeft)
 			{
 				var rowData = new List<byte>();
 				for (var x = 0; x < header.Width; x++)
-					rowData[x] = DEFAULT_INDEX;
+					rowData.Add(DEFAULT_INDEX);
```

Comment:
This fills rows with the `DEFAULT_INDEX` which is not desired.
We only want to define data that actually exists.